onnx/defs/tensor/utils.cc

```diff
@@ -56,6 +56,61 @@ void KeepAspectRatioHelper(
   }
 }
 
+void gridSampleShapeInference(InferenceContext& ctx) {
+  propagateElemTypeFromInputToOutput(ctx, 0, 0);
+
+  // Grid sample input tensor indices.
+  size_t const input_param = 0, grid_param = 1;
+
+  auto const& input_shape = getInputShape(ctx, input_param);
+  auto const& grid_shape = getInputShape(ctx, grid_param);
+
+  if (input_shape.dim_size() != grid_shape.dim_size()) {
```

Comment:
Need to add
```cpp
if (!hasNInputShapes(2)) return;
```
before this to ensure that we skip the tests if the input shapes are unknown.

Reply:
Done.